### DISCUSSION TITLE
how to upload/add upstream sever's root certificate

### DISCUSSION THREAD

**Discussion by Asilent** [Kong/kong#7809] [Help] [answered]
I want to call upstream server using self-sign certificate with https. 
I found kong can support https by uploading crt&private key. 
But I feel that this is a solution that provides https for the kong itself.

**Comment by liyangau** [comment, marked as answer]
Please clarify what you are trying to do. 

1. Use self-sign `client certificate` to do mTLS with upstream server.
OR
2. Upstream server is using a self-sign `SSL` and you want to trust this self-sign certificate.
